### DISCUSSION TITLE
Interpretation of sensitivity function values

### DISCUSSION THREAD

**Discussion by shoepfl** [AMICI-dev/AMICI#1786] [Q&A] [answered]
I have a question about the actual implementation of the forward sensitivity analysis. I assume amici calculates the forward sensitivity like:

![grafik](https://user-images.githubusercontent.com/66028655/164504750-f0e0414f-c02f-48f7-ba82-e41de0fc701c.png)

with the kth parameter of the parameter vector theta and the state vectors xi. This would however not allow to compare the values of different species with different units or different units of parameters right? So is there any normalization included?

Thanks for the feeback,

Sebastian

**Comment by FFroehlich** [comment, marked as answer]
> I have a question about the actual implementation of the forward sensitivity analysis. I assume amici calculates the forward sensitivity like:
> 
> ![grafik](https://user-images.githubusercontent.com/66028655/164504750-f0e0414f-c02f-48f7-ba82-e41de0fc701c.png)
> 
> with the kth parameter of the parameter vector theta and the state vectors xi. This would however not allow to compare the values of different species with different units or different units of parameters right? So is there any normalization included?
> 
> Thanks for the feeback,
> 
> Sebastian

Hi Sebastian. Yes you are correct, AMICI always computes unnormalized sensitivities. For most biochemical systems, I would recommend  calculating sensitivities with respect to a logarithmic scale as this is a more natural scale (for mass action kinetics log-parameters correspond to Gibbs free energies) and should also address any concerns about normalization or units.

Hope this helps,
Fabian

**Reply by shoepfl** [reply]
Thanks for the fast reply Fabian,

yes this helps. So log transformation of the parameters with
`amici_model.setParameterScale(amici.ParameterScaling.log10)`
should do the job to get normalized sensitivities, right?

Best,

Sebastian

**Reply by FFroehlich** [reply]
Yes, thats correct. `setParameterScale` expects a vector of value though, one for each parameter in the same ordering as `getParameterIds` (It looks like the documentation for that function is not being exported correctly)/
